### DISCUSSION TITLE
weak functions & command mode

### DISCUSSION THREAD

**Discussion by danilkorotkov** [HomeSpan/HomeSpan#65] [General]
HomeSpan has command modes: the button is pressed or received Serial command
some user apps require (as your Somfy) to create, update or format own NVS storage independently of HomeSpan storage.

It would be great to reserve at least one user letter and button :
```
    case 6: //user reserved
      call weak func to process command;
    break;  
```

serial:
1
```
 case '?': {
...
weak func for serial output
...
}
```
2
```
case 'user letter': {
call weak func to process command;
}
```

**Comment by HomeSpan** [comment]
Good idea - I'll actually make it more generic so you can register any number of command/function combinations, including a description that will add to the '?' list.  The functions will need to accept a single char* pointer to retrieve any data.   I'll also document the hook into my serial overlay that makes it easy to read a max number of characters from the serial port.
